tests/wallet/vault/test_vault_wallet.py

```diff
@@ -1,19 +1,82 @@
 from __future__ import annotations
 
+import types
 from hashlib import sha256
+from typing import Awaitable, Callable, Type
 
 import pytest
 from ecdsa import NIST256p, SigningKey
 
 from chia.util.ints import uint32, uint64
+from chia.util.observation_root import ObservationRoot
 from chia.wallet.util.tx_config import DEFAULT_TX_CONFIG
+from chia.wallet.vault.vault_root import VaultRoot
+from chia.wallet.vault.vault_wallet import Vault
+from chia.wallet.wallet_protocol import MainWalletProtocol
+from chia.wallet.wallet_state_manager import WalletStateManager
 from tests.conftest import ConsensusMode
 from tests.wallet.conftest import WalletStateTransition, WalletTestFramework
 
 SECP_SK = SigningKey.generate(curve=NIST256p, hashfunc=sha256)
 SECP_PK = SECP_SK.verifying_key.to_string("compressed")
 
 
+async def vault_setup(
+    wallet_environments: WalletTestFramework, monkeypatch: pytest.MonkeyPatch, with_recovery: bool
+) -> None:
+    def get_main_wallet_driver(self: WalletStateManager, observation_root: ObservationRoot) -> Type[MainWalletProtocol]:
+        return Vault
+
+    monkeypatch.setattr(
+        WalletStateManager,
+        "get_main_wallet_driver",
+        types.MethodType(get_main_wallet_driver, WalletStateManager),
+    )
```

Comment:
Don't monkeypatch this like in the ACS test.  You can actually change the function in WSM to properly return Vault based on the observation_root.  I only monkeypatched it in the ACS test because that wallet shouldn't be part of the main codebase.